### PR TITLE
Kkrol grid suggestions

### PR DESCRIPTION
**Pull request by Krystian-Krol**
# Description
Tested on LocalBackend, DGX and Entropy (entropy failed due to dataset lock, which is after the scope of this PR)
[Neptune for DGX and Entopy](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=9ac46a51-c84e-442d-963c-ac6a3a8af3f3&detailsTab=metadata&shortId=LLMRANDOM-7675&type=run&compare=GwDgDA7ANATFCMBmMUCsBOBX7aA&lbViewUnpacked=true&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=true&query=((%60sys%2Fowner%60%3Astring%20%3D%20%22kkrol%22))%20AND%20(%60sys%2Fid%60%3Astring%20CONTAINS%20%22LLMRANDOM-767%22))

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/grid/setup_arguments.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import os
+from typing import List, Optional
+
+
+def make_singularity_mount_paths(setup_args: dict, training_args: dict) -> str:
```
Comment (crewtool) [inline lizrd/grid/setup_arguments.py:5]: Może wrzucić to wewnątrz klasy MachineBackend jako statyczną metodę, bo, tylko tam będzie używane + zmienić, żeby przyjmowało hf_datasets_cache, train_dataset_path i validation_dataset_path zamiast słowników

File: lizrd/grid/setup_arguments.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    )
+    singularity_mount_paths += (
+        f",{training_args['validation_dataset_path']}:{training_args['validation_dataset_path']}"
+        if training_args["validation_dataset_path"] is not None
+        else ""
+    )
+    return singularity_mount_paths
+
+
+def make_singularity_env_arguments(
```
Comment (crewtool) [inline lizrd/grid/setup_arguments.py:29]: To też wrzucić do klasy

File: lizrd/grid/infrastructure.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    @classmethod
+    def get_subprocess_args(
+        cls,
+        slurm_command,
+        setup_args,
+        training_args,
+        singularity_env_arguments,
+        singularity_mount_paths,
+        runner_params,
+    ):
```
Comment (crewtool) [inline lizrd/grid/infrastructure.py:92]: Po przerzuceniu funkcji budujących te argumenty do klasy, można je będzie zbudować tutaj
```suggestion
        runner_params,
    ):
        singularity_mount_paths = self.make_singularity_mount_paths(...)
        singularity_env_arguments = self.make_singularity_env_arguments(...)
```

File: lizrd/grid/infrastructure.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    @classmethod
+    def get_grid_entrypoint(cls) -> str:
+        return "lizrd/scripts/grid_entrypoint.sh"
+
+    @classmethod
+    def get_subprocess_args(
+        cls,
+        slurm_command,
+        setup_args,
+        training_args,
```
Comment (crewtool) [inline lizrd/grid/infrastructure.py:88]: Chyba job_name jest z tego tylko użyte, więc można przerobić

File: lizrd/grid/infrastructure.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+        singularity_env_arguments,
+        singularity_mount_paths,
+        runner_params,
```
Comment (crewtool) [inline lizrd/grid/infrastructure.py:91]: Tutaj ewentualnie myślałem o rozdzieleniu budowania infrastruktury od parametrów, ale można w osobnej PR

File: lizrd/grid/infrastructure.py (hunk 3 above)
Comment (Krystian-Krol) [reply lizrd/grid/infrastructure.py:92]: nie mam `neptune_key` ani `wandb_key` więc tylko mount_paths moge przerzucić

File: lizrd/grid/infrastructure.py (hunk 4 above)
Comment (Krystian-Krol) [reply lizrd/grid/infrastructure.py:88]: już nie skoro usunąłem `mount_path`

File: lizrd/scripts/grid.py [hunk 6]
```diff
@@ -43,16 +41,16 @@ def create_subprocess_args(
...
     skip_copy_code=False,
 ):
-    configs = prepare_configs(config_path, git_branch, CLUSTER_NAME)
+    configs = prepare_configs(config_path, git_branch, CLUSTER)
     grid = setup_experiments(configs)
     check_for_argparse_correctness(grid)
     interactive_debug_session = grid[0][0]["interactive_debug_session"]
 
-    if CLUSTER_NAME != MachineBackend.LOCAL and not skip_confirmation:
+    if not isinstance(CLUSTER, LocalBackend) and not skip_confirmation:
```
Comment (Krystian-Krol) [inline lizrd/scripts/grid.py:53]: nie jestem fanem tych patternów, powinienem wywalić je do metody Backendu? Jeśli tak, to czy trzymać w Backendzie  `args.skip_confirmation`?

**Review by crewtool — APPROVED** [review]
Nice! LGTM

**Pull request merged**